gph/src/ripser.h

```diff
@@ -1298,12 +1290,12 @@ class ripser
 
         // extra vector is a work-around inability to store floats in the
         // hash_map
-        std::atomic<size_t> last_diameter_index{0}, idx_essential{0};
-        entry_hash_map deaths;
-        deaths.reserve(columns_to_reduce.size());
+        std::atomic<size_t> idx_persistence{0}, idx_essential{0};
```

Comment:
`last_diameter_index` makes me think that this stores the edge indices of the largest edge within a simplex.  Is this correct?

Reply:
`last_diameter_index` was really a bad name.
During the computation, this was storing the number of finite barcodes in the current dimension.
We were incrementing this value each time a new finite barcode was found, and we use the pre incremented value as an index to store in another vector the death value.

This is currently a bit messy, but it is due to a current limitation of the hash map to store floating values.
I hope that at some point we will do a new concurrent lock-free hash_map that supports float values.

Reply:
Thanks for the explanation!  `What would you think about `idx_finite_bar` for this variable?  Because `persistence` for me is a death - birth value.

Reply:
I am all good for suggestions, it is a draft to try to improve variable names.
Any feedback from experts (like you ;) ) in the field is more than welcome !

Reply:
Also, to align with another variable, what do you think about changing `new_bar_pers_idx` to `new_finite_idx` ?

Reply:
> Also, to align with another variable, what do you think about changing new_bar_pers_idx to new_finite_idx ?

I agree about the need to align, but then I would suggest new_idx_finite_bar if we go for idx_finite_bar.

Reply:
Good for me, I will push an update and if everything else is good for you, I think we can merge !

Reply:
If good for you, I let you resolve the conversation

Reply:
Thanks!